sounds/models.py

```diff
@@ -730,9 +730,12 @@ def post_delete_comment(self, commit=True):
             self.save()
 
     def compute_crc(self, commit=True):
-        with open(self.locations('path'), "rb") as fp:
-            data = fp.read()
-        crc = zlib.crc32(data)
+        crc = 0
+        with open(self.locations('path'), 'rb') as fp:
+            data = fp.read(settings.CRC_BUFFER_SIZE)
+            while len(data) > 0:
+                crc = zlib.crc32(data, crc)
+                data = fp.read(settings.CRC_BUFFER_SIZE)
         self.crc = '{:0>8x}'.format(crc & 0xffffffff)  # right aligned with zero-padding, width of 8 chars
```

Comment:
I find this breaks the DRY principle... If you don't want to use the function iteration example I gave, then perhaps:

```python
while True:
        chunk = f.read(4096)
        if not chunk:
            break
        dostuff(chunk)
```

Makes more sense

Reply:
I wanted to make code a bit more understandable, but I can see your point. And `iter` is much better than `break` from `while True`.